core-build/gulp-core-build/src/tasks/JestReporter.ts

```diff
@@ -30,31 +22,29 @@ class JestReporter extends (DEFAULT_REPORTER as { new (globalConfig: Jest.Global
   public onRunComplete(contexts: Set<Jest.Context>, results: Jest.AggregatedResult): void {
     super.onRunComplete(contexts, results);
 
-    let outputPath: string | undefined = undefined;
-    let filename: string | undefined = undefined;
-
+    // Since multiple reporters can be used, we need to look through the reporters list to
+    // find this one. We also want to check if writing the output file was enabled on the
+    // configuration and that we have an output file path
     for (const reporter of (this._globalConfig.reporters as (string | ReporterConfig)[])) {
```

Comment:
It looks like the Jest reporter API provides the reporter options in the second constructor argument: https://github.com/facebook/jest/blob/master/packages/jest-reporters/src/coverage_reporter.ts#L32-L35

I think we can probably simplify this logic quite a bit by using that.

Reply:
Good catch!